maturity-level-two/docs/api-security.md

```diff
@@ -0,0 +1,127 @@
+# API Security
+API-security is an important topic on design & development time of the API.
+
+The document below is explaining the general guidelines and goes into more detail in the second topic specifically for Azure solutions.
+
+1. [General Guidelines](#general-http-guidelines)
+    - [HTTPS](#HTTPS)
+    - [IP-Filtering](#IP-Filtering)
+    - [API Key (Shared Access Key)](#API-Key-(Shared-Access-Key))
+    - [TLS Mutual Authentication](#TLS-Mutual-Authentication)
+    - [OAuth2 Token (From Identity Provider)](#OAuth2-Token-(From-Identity-Provider))
+   
+2. [Azure Specifics](#Azure-specifics)
+   - [API Management](#API-Management)
+   - [Web App](#Web-App)
+   - [Managed Service Identity](#Managed-Service-Identity)
+   - [Azure Functions](#Azure-Functions)
+
+
+## General Guidelines
+Security can be done on different levels and all have their up & downsides.
+This guideline discuss's first the least secure solution & goes up based on that. Please take that into consideration when choosing your security mechanism.  
+You can also combine all of these guidelines, you do not need to stick to one.
+
+### HTTPS
+HTTPS and only HTTPS should be allowed on production API's. This is mandatory security measurement to take when designing and creating your API.
+
+Please make sure that you have a valid server certificate and that it remains valid.
+It is possible to bypass server certificate validation on the client side, but this should never be done in any case!
+
+### IP-Filtering
+IP-Filtering is a system where you validate the incoming IP-address on the server. This can be one item or a list. Preferably, this list is as small as possible. 
+
+Note that this does require a static IP of all the users, and by this validation, still ALL machines behind that public-IP can access the resource.
+
+IP-filtering is done ideally on firewall level/API-Management.
+
+### API Key (Shared Access Key)
+The API-Key is a custom HTTP Header with a specific name. Usually this is called `x-API-key`, but can also be called otherwise if required.
+
+The value of this key should be shared between the user and the server.
+The server / API will validate the incoming request via an `Authentication Filter` and verify the value matches between the user and the stored one on the API.
+
+To increase security you can here also use `Rolling API keys` where you allow 2 keys at one time, and you require all the users to update to the new key periodically. This way, if a APIKey is intercepted, it will only be valid for x-amount of time. ALso, having just the mecahnism in place and being able to force the rolling in a specific time can be beneficial to fix unwanted resource access.
```

Comment:
```suggestion
To increase security you can here also use `Rolling API keys` where you allow 2 keys at one time, and you require all the users to update to the new key periodically. This way, if a APIKey is intercepted, it will only be valid for a certain amount of time. Also, having just the mechanism in place and being able to force the rolling at a specific time can be beneficial to fix unwanted resource access.
```